teps/0044-composing-tasks-with-tasks.md

```diff
@@ -0,0 +1,173 @@
+but the PVCs still slow down the overall execution.
+
+### PipelineResources
+
+This kind of compositional functionality was being (somewhat! read on!) provided
+by [PipelineResources](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md).
+[We did not bring these types to beta](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#why-arent-pipelineresources-in-beta).
+Issues that don't let PipelineResources (as is) solve these problems are:
+* PipelineResource "outputs" do not run when the steps of the Task fail (or if
+  a previous PipelienResource "output" fails) (see [unit test use case](#use-cases-optional))
+* PipelineResources do not use Tasks (which is something we could change), meaning
+  you cannot use them to compose Tasks together, you need to build PipelineResources
+  * Adding new PipelineResources currently needs to be done in the Tekton Pipelines controller,
+    tho there have been several attempts to propose ways to make it extensible:
+    * [PipelineResources 2 Uber Design Doc](https://docs.google.com/document/d/1euQ_gDTe_dQcVeX4oypODGIQCAkUaMYQH5h7SaeFs44/edit)
+    * [Tekton Pipeline Resource Extensibility](https://docs.google.com/document/d/1rcMG1cIFhhixMSmrT734MBxvH3Ghre9-4S2wbzodPiU/edit)
+  * All of this is even though PipelineResources are _very_ similar to Tasks in that they are
+    collections of steps (which [get injected before and after a Task's steps](https://github.com/tektoncd/pipeline/issues/1838)
+* Tasks have to declare in advance what PipelineResources they need; you can't decide to use
+  a PipelineResource with a Task after it has been written (e.g. in a TaskRun or a Pipeline)
+  and you can't mix types of PipelineResource (e.g. if a Task declares it needs a
+  [git PipelineResources](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#git-resource)
+  it can't use a different kind of PipelineResource to provide those files (tho you can avoid
+  doing the clone multiple times using [from](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-the-from-parameter))
+
+### Goals
+
+- Make it possible to combine Tasks together so that you can run multiple
+  Tasks as "one unit" (see [Requirements](#requirements) at authoring time
+  in a way that can be reused (e.g. in a Pipeline)
+- Add some of [the features we don't have without PipelineResources](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#)
+  to Tekton Pipelines (without requiring use of PipelineResources), specifically **Task adapters/specialization**
+  - If possible we can provide a migration path from PipelineResources to the solution proposed here, where applicable
+
+### Non-Goals
+
+- Composing Tasks within Tasks at [Task authoring time](https://github.com/tektoncd/community/blob/master/design-principles.md#reusability).
```

Comment:
So one thing that bothers me now is : "Composing Tasks within Tasks" is part of the non-goal **but** the name of the TEP is literraly : _Composing Tasks with Tasks_.

Reply:
the non-goal is the "at task authoring time" part, the goal is still to compose tasks with tasks, just not while writing at ask

Reply:
"authoring time" could be when authoring a Task or when authoring a Pipeline. I'm trying to say that this scope of this TEP is _Pipeline Authoring time_, but note _Task authoring time_

Reply:
I was also trying to leave it vague in order to be open to the possibility that what we're missing is some other unit of composition - the point of this non-goal was to say that we don't want to use tasks with tasks inside of the Task CRD

Reply:
anyway ill try to completely reword this to be more clear